articles/osquery-evented-tables-overview.md

```diff
@@ -0,0 +1,161 @@
+- [What are osquery evented tables?](#what-are-osquery-evented-tables?)
+- [How do evented tables work?](#how-do-evented-tables-work?)
+- [What do I need to consider when configuring evented tables?](#what-do-i-need-to-consider-when-configuring-evented-tables?)
+- [How do I turn on an evented table?](#how-do-i-turn-on-an-evented-table?)
+-- [File integrity monitoring](#file-integrity-monitoring-fim)
+-- [Process auditing](#process-auditing)
+-- [Yara scanning](#yara-scanning)
+
+# Osquery evented tables
+
+## What are osquery evented tables? 
+Osquery traditionally allows you to see the state of your computers right now. This point-in-time snapshot, however, requires you to actively watch the table repeatedly. For example, in the security-critical `processes` table, a process might be very short-lived and there is high churn in processes. Actively watching and diffing this table would be challenging even with automation.
+
+This is where osquery's evented tables come in. Rather than displaying the point-in-time state for your host, osquery's evented tables store the host's historical data. You can configure osquery to capture certain types of information, after which it will be stored in the relevant `*_events` table for later analysis. 
+
+This guide will walk you through the concepts, considerations, and best practices for setting up evented tables, in particular focusing on osquery. It will also provide basic information about commonly-used evented tables for a newcomer to get started.
+
+## How do osquery evented tables work?
+Osquery does not generate the events itself. Instead, it reads and formats event data generated by system utilities. For example, on Linux, audit generates process and socket event data in its own storage location and format. Osquery reads data from audit’s storage location, converts it into an event row, and buffers them (in the internal [rocksdb](http://rocksdb.org) store) in the `process_events` and `socket_events` tables to await querying. The data can then be filtered and transformed via SQL, and shipped to a log destination with the scheduled query functionality. 
```

Comment:
> audit generates process and socket event data in its own storage location and format

I am not the most well versed with audit on Linux, but is this necessarily true? I think it's more accurate to say something like `audit kernel component collects these records/messages and sends it to the auditd daemon which may create a log entry`, or we can skip the explanation and say something along the lines of `it uses audit under the hood`

Reply:
You're right. There's a difference between audit and auditd. Auditd will collect the data in some location. Audit simply broadcasts it. Will update.